apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -534,4 +541,97 @@ export const pageRouter = router({
       }
       return permalinkTree
     }),
+
+  createIndexPage: protectedProcedure
+    .input(createIndexPageSchema)
+    .mutation(async ({ ctx, input: { siteId, parentId } }) => {
+      await validateUserPermissionsForResource({
+        userId: ctx.user.id,
+        siteId: siteId,
+        action: "create",
+        resourceId: parentId,
+      })
+
+      // Validate whether parentId exists and is a Folder or Collection
+      const parent = await db
+        .selectFrom("Resource")
+        .where("Resource.id", "=", parentId)
+        .where("Resource.siteId", "=", siteId)
+        .where("Resource.type", "in", [
+          ResourceType.Folder,
+          ResourceType.Collection,
+        ])
+        .select(["title", "permalink", "type"])
+        .executeTakeFirst()
+
+      if (!parent) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Parent resource not found or is not a folder/collection",
+        })
+      }
+
+      // Get the full permalink of the parent
+      const parentFullPermalink = await getResourceFullPermalink(
+        siteId,
+        parseInt(parentId),
+      )
+      if (!parentFullPermalink) {
+        throw new TRPCError({
+          code: "NOT_FOUND",
+          message: "Could not determine parent's full permalink",
+        })
+      }
+
+      return db.transaction().execute(async (tx) => {
+        const blob = await tx
+          .insertInto("Blob")
+          .values({
+            content: jsonb({
+              page: {
+                title: parent.title,
+                permalink: parentFullPermalink,
+                lastModified: new Date().toISOString(),
+                contentPageHeader: {
+                  // follow the same format as autogenerated index pages
+                  summary: `Pages in ${parent.title}`,
+                },
+              },
+              layout:
+                parent.type === ResourceType.Collection
+                  ? ISOMER_USABLE_PAGE_LAYOUTS.Collection
+                  : ISOMER_USABLE_PAGE_LAYOUTS.Index,
```

Comment:
clarification - shouldn't this be `Index`?

Reply:
i looked through existing custom index pages for collections and they are all Collection

Reply:
i doubled checked with existing collextion indexpages and have slightly tweaked it in https://github.com/opengovsg/isomer/pull/1065/commits/f8a14bb6f63a4141c85e5a345b8ac307c9aea30e - thanks for catching